### PR TITLE
Logging of information about the current system

### PR DESCRIPTION
**Pull request by Crystal15118**
Information about the system is printed to the log file when client is initialized.

### CODE REVIEW

File: src/main/java/me/contaria/seedqueue/SeedQueue.java [hunk 1]
```diff
@@ -71,6 +72,19 @@ public void onInitializeClient() {
         }
     }
 
+    public static void logSystemInformation() {
+        LOGGER.info("System Information:");
+        LOGGER.info("Operating System: {}", System.getProperty("os.name"));
+        LOGGER.info("OS Version: {}", System.getProperty("os.version"));
+        LOGGER.info("CPU: {}", System.getenv("PROCESSOR_IDENTIFIER"));
+        LOGGER.info("Available Processors (Cores): {}", Runtime.getRuntime().availableProcessors());
```
[the remark this answers is not in the document]
Comment (tesselslate) [reply src/main/java/me/contaria/seedqueue/SeedQueue.java]: IIRC it is indeed affected by affinity, at least on Linux. I had Minecraft crash logs saying I had only 2 processors despite having 24 due to affinity.

File: src/main/java/me/contaria/seedqueue/SeedQueue.java [hunk 2]
```diff
@@ -71,6 +73,27 @@ public void onInitializeClient() {
         }
     }
 
+    private static void logSystemInformation() {
+        int availableProcessors = Runtime.getRuntime().availableProcessors();
+        int totalProcessors = getTotalProcessors();
```
Comment (KingContaria) [inline src/main/java/me/contaria/seedqueue/SeedQueue.java:78]: Could you just inline these and remove the `getTotalProcessors` method?

File: src/main/java/me/contaria/seedqueue/SeedQueue.java [hunk 3]
```diff
@@ -71,6 +73,27 @@ public void onInitializeClient() {
...
+
+        LOGGER.info("System Information (Logged by SeedQueue):");
+        LOGGER.info("Operating System: {}", System.getProperty("os.name"));
+        LOGGER.info("OS Version: {}", System.getProperty("os.version"));
+        LOGGER.info("CPU: {}", System.getenv("PROCESSOR_IDENTIFIER"));
+        LOGGER.info("Java Version: {}", System.getProperty("java.version"));
+        LOGGER.info("JVM Arguments: {}", String.join(" ", java.lang.management.ManagementFactory.getRuntimeMXBean().getInputArguments()));
+        LOGGER.info("Allocated Memory (MB): {}", Runtime.getRuntime().totalMemory() / (1024 * 1024));
+        LOGGER.info("Max Memory (MB): {}", Runtime.getRuntime().maxMemory() / (1024 * 1024));
+        LOGGER.info("Free Memory (MB): {}", Runtime.getRuntime().freeMemory() / (1024 * 1024));
```
Comment (KingContaria) [inline src/main/java/me/contaria/seedqueue/SeedQueue.java:88]: Correct me if I'm wrong, but `Runtime#totalMemory` returns the amount of RAM used at that very moment, which isn't useful and `Runtime#freeMemory` also isn't helpful.
Max Memory seems to be Allocated Memory and if there is some kind of way to get the systems total memory through `ManagementFactory` that would be helpful information

File: src/main/java/me/contaria/seedqueue/SeedQueue.java [hunk 4]
```diff
@@ -71,6 +73,27 @@ public void onInitializeClient() {
...
+    private static void logSystemInformation() {
+        int availableProcessors = Runtime.getRuntime().availableProcessors();
+        int totalProcessors = getTotalProcessors();
+
+        LOGGER.info("System Information (Logged by SeedQueue):");
+        LOGGER.info("Operating System: {}", System.getProperty("os.name"));
+        LOGGER.info("OS Version: {}", System.getProperty("os.version"));
+        LOGGER.info("CPU: {}", System.getenv("PROCESSOR_IDENTIFIER"));
+        LOGGER.info("Java Version: {}", System.getProperty("java.version"));
+        LOGGER.info("JVM Arguments: {}", String.join(" ", java.lang.management.ManagementFactory.getRuntimeMXBean().getInputArguments()));
```
Comment (KingContaria) [inline src/main/java/me/contaria/seedqueue/SeedQueue.java:85]: No need to specify `java.lang.management.ManagementFactory`, just `ManagementFactory` works.

**Review by KingContaria — CHANGES_REQUESTED** [review]

File: src/main/java/me/contaria/seedqueue/SeedQueue.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Crystal15118) [reply src/main/java/me/contaria/seedqueue/SeedQueue.java]: > IIRC it is indeed affected by affinity, at least on Linux. I had Minecraft crash logs saying I had only 2 processors despite having 24 due to affinity.

fixed :)

**Review by KingContaria — COMMENTED** [review]
Lgtm, please make a test build and have atleast one person using mac, linux and windows test if it prints the correct values

**Comment by maskersss** [comment]
> `Runtime#totalMemory` returns the amount of RAM used at that very moment, which isn't useful and `Runtime#freeMemory` also isn't helpful

is it not? would something like "current amount of free ram + current amount of ram used by minecraft" not be a good way to judge how much ram you can allocate? obviously there's going to be some rng in how much ram everything is using, but it still sounds useful to me
the only issue i can maybe see with it is if OS is paging some memory, idk if it's possible to figure it out, but there could be more issues i haven't thought of

**Comment by KingContaria** [comment]
`Runtime#maxMemory` gives the amount of allocated ram

**Comment by maskersss** [comment]
oh i think i misunderstood what this means, is `Runtime#totalMemory` / `Runtime#freeMemory` referring to the amount of total/free memory used by the java process? i thought it was about ram on your pc in general mb
in this case, still, would it be useful/possible/has it been implemented to check the amount of free ram on your pc (accounting for obs and stuff)?

**Comment by KingContaria** [comment]
It could be helpful in some cases but misleading in others because from reading the log we can't know if they started OBS before or after Minecraft

**Comment by tildejustin** [comment]
could you add a system property to disable this? I didn't like sleepbg config printing so I'd rather be able to turn this off as well and be able to read my logs easier.

**Comment by Crystal15118** [comment]
> could you add a system property to disable this? I didn't like sleepbg config printing so I'd like to be able to turn this off as well and be able to read my logs easier.

i can but it will likely take much more time and i don't think it would be as worth it as the effort but let's see what others say as well

**Comment by KingContaria** [comment]
I'm not opposed to a system property

**Review by KingContaria — APPROVED** [review]
Lgtm, thank you :)

**Pull request merged**